packages/nx/src/command-line/add/add.ts

```diff
@@ -0,0 +1,186 @@
+
+    await installPackage(pkgName, version, pmc);
+    await initializePlugin(pkgName, pmc);
+
+    output.success({
+      title: `Package ${pkgName} added successfully.`,
+    });
+  });
+}
+
+async function installPackage(
+  pkgName: string,
+  version: string,
+  packageManagerCommands: PackageManagerCommands
+): Promise<void> {
+  const spinner = ora(`Installing ${pkgName}@${version}...`);
+  spinner.start();
+
+  if (existsSync('package.json')) {
+    await new Promise<void>((resolve) =>
+      exec(
+        `${packageManagerCommands.addDev} ${pkgName}@${version}`,
+        (error, stdout) => {
+          if (error) {
+            spinner.fail();
+            output.addNewline();
+            logger.log(stdout);
+            output.error({
+              title: `Failed to install ${pkgName}. Please check the error above for more details.`,
+            });
+            process.exit(1);
+          }
+
+          return resolve();
+        }
+      )
+    );
+  } else {
+    const nxJson = readNxJson();
+    nxJson.installation.plugins ??= {};
+    nxJson.installation.plugins[pkgName] = version;
+    writeJsonFile('nx.json', nxJson);
+
+    const cwd = process.cwd();
+    const offsetFromRoot = relative(cwd, workspaceRootInner(cwd, null));
+
+    let nxExecutable: string;
+    if (process.platform === 'win32') {
+      nxExecutable = '.\\' + join(`${offsetFromRoot}`, 'nx.bat');
+    } else {
+      nxExecutable = './' + join(`${offsetFromRoot}`, 'nx');
+    }
+
+    await new Promise<void>((resolve) =>
+      exec(`${nxExecutable}`, (error, _stdout, stderr) => {
+        if (error) {
+          // revert adding the plugin to nx.json
+          nxJson.installation.plugins[pkgName] = undefined;
+          writeJsonFile('nx.json', nxJson);
+
+          spinner.fail();
+          output.addNewline();
+          logger.log(stderr);
+          output.error({
+            title: `Failed to install ${pkgName}. Please check the error above for more details.`,
+          });
+          process.exit(1);
+        }
+
+        return resolve();
+      })
+    );
+  }
+
+  spinner.succeed();
+}
+
+async function initializePlugin(
+  pkgName: string,
+  packageManagerCommands: PackageManagerCommands
+): Promise<void> {
+  const capabilities = await getPluginCapabilities(workspaceRoot, pkgName, {});
+  const generators = capabilities?.generators;
+  if (!generators) {
+    output.log({
+      title: `No generators found in ${pkgName}. Skipping initialization.`,
+    });
+    return;
+  }
+
+  const initGenerator = findInitGenerator(generators);
+  if (!initGenerator) {
+    output.log({
+      title: `No "init" generator found in ${pkgName}. Skipping initialization.`,
+    });
+    return;
+  }
+
+  const spinner = ora(`Initializing ${pkgName}...`);
+  spinner.start();
+
+  await new Promise<void>((resolve) =>
+    exec(
+      `${packageManagerCommands.exec} nx g ${pkgName}:${initGenerator}`,
```

Comment:
There's a util somewhere thats something like `runNxSync` that we should use here, it handles the pmc and encapsulated nx: https://github.com/nrwl/nx/blob/master/packages/nx/src/utils/child-process.ts#L7C20-L7C20

Reply:
Yeah, I saw it before, but I didn't use it because we need an async operation, so the `ora` spinner animation renders. Still, it makes sense to keep the related logic in the same place, so I added now an async util in that file to do this.

Reply:
I think this still needs to be updated to use the runNxAsync function?